### PR TITLE
Added CreateUserAccount endpoint and added permissions checker

### PR DESCRIPTION
**Pull request by chris-bes**
### Changes:

- Added generic CreateHandler and CreateUserAccount handler

More context: https://github.com/beyondessential/tupaia-backlog/issues/1028#issuecomment-701270414

### CODE REVIEW

File: packages/meditrak-server/src/routes/createUser.js [hunk 1]
```diff
@@ -24,8 +24,9 @@ const PERMISSION_GROUPS = {
 
 const DEMO_LAND_NAME = 'Demo Land';
 
+//Used for registering user account
 export const createUser = async (req, res) => {
-  await req.assertPermissions(assertBESAdminAccess);
+  await req.assertPermissions(allowNoPermissions);
```
Comment (chris-bes) [inline packages/meditrak-server/src/routes/createUser.js:29]: This endpoint is actually for registering user account from Tupaia webpage so I think this should actually be for "anyone"

**Review by edmofro — COMMENTED** [review]
I see @EMcQ-BES has picked this up for review, but my question after a brief look is whether the two create user handlers could share more code? Perhaps the one servicing tupaia.org could extend the other, but put more restrictions on what data is allowed to be sent through (so they can't specify their own email verified status, for example)?

File: packages/meditrak-server/src/routes/createUser.js [hunk 2]
```diff
@@ -24,8 +24,9 @@ const PERMISSION_GROUPS = {
 
 const DEMO_LAND_NAME = 'Demo Land';
 
+//Used for registering user account
 export const createUser = async (req, res) => {
```
Comment (EMcQ-BES) [inline packages/meditrak-server/src/routes/createUser.js:28]: Should we convert this to a CreateHandler to keep consistent?

File: packages/meditrak-server/src/app/addRoutesToApp.js [hunk 3]
```diff
@@ -153,6 +154,7 @@ export function addRoutesToApp(app) {
   app.post('(/v[0-9]+)/import/users', upload.single('users'), importUsers);
   app.post('(/v[0-9]+)/import/optionSets', upload.single('optionSets'), importOptionSets);
   app.post('(/v[0-9]+)?/user', createUser);
+  app.post('(/v[0-9]+)?/userAccount', createUserAccount);
```
Comment (EMcQ-BES) [inline packages/meditrak-server/src/app/addRoutesToApp.js:157]: Might be nice to have a comment explaining the difference between these end points. Would be confusing if you're reading this without context.

**Review by EMcQ-BES — APPROVED** [review]
Love it! I appreciate the separate verbiage to make a distinction between the endpoints.

**Pull request merged**